hcl/eval/partial.go

```diff
@@ -216,13 +216,33 @@ func (e *engine) emitn(n int) {
 	}
 }
 
-func (e *engine) emitVariable(v variable) {
+func (e *engine) emitVariable(v variable) error {
 	tos := e.evalstack.peek()
+	for i, original := range v.index {
+		// Prepare a subengine to evaluate the indexing tokens.
+		// The partial eval engine expects the token stream to be EOF terminated.
+		index := make(hclwrite.Tokens, len(original))
```

Comment:
Don't we have a copy tokens function ? (I remember seeing it in a previous PR)

Reply:
yeah we have.. in this case we dont need a full copy but just a new slice with the token pointers (they will be copied anyway in the PartialEval constructor).
not sure what to do yet, I'll try to avoid more copies of the tokens.
let me think more about this.